src/AccentColor/ColorExtractor.vala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+ */
+
+public class Gala.ColorExtractor : Object {
+    private const double PERCENTAGE_SAMPLE_PIXELS = 0.01;
+
+    public Gdk.Pixbuf pixbuf { get; construct set; }
+
+    private Gee.List<Granite.Drawing.Color> pixels;
+
+    public ColorExtractor (Gdk.Pixbuf pixbuf) {
+        Object (pixbuf: pixbuf);
+
+        pixels = convert_pixels_to_rgb (pixbuf.get_pixels_with_length (), pixbuf.has_alpha);
+    }
+
+    public int get_dominant_color_index (Gee.List<Granite.Drawing.Color> palette) {
```

Comment:
This runs **really** fast on my PC... But I'm on a Ryzen 9 so :man_shrugging: 

**If** it is slow on lower end devices, maybe this could be performed once when the wallpaper is imported, saved in the EXIF data and use here.

Reply:
As you can see in the video above, calculating and setting the accent color is pretty fast even on my low-power hardware. The accent color is even set before the background image has been graphically replaced. 

![Bildschirmfoto von 2021-04-11 15 31 38](https://user-images.githubusercontent.com/10796736/114306246-a879c500-9adb-11eb-8ae2-7dc90e0c3ecb.png)

Reply:
Awesome :+1: